rust/lance-encoding/src/decoder.rs

```diff
@@ -1167,7 +1173,17 @@ impl BatchDecodeStream {
     fn task_to_batch(task: NextDecodeTask) -> Result<RecordBatch> {
         let struct_arr = task.task.decode();
         match struct_arr {
-            Ok(struct_arr) => Ok(RecordBatch::from(struct_arr.as_struct())),
+            Ok(struct_arr) => {
+                let batch = RecordBatch::from(struct_arr.as_struct());
+                let size_bytes = batch.get_array_memory_size() as u64;
+                if size_bytes > BATCH_SIZE_BYTES_WARNING {
+                    EMITTED_BATCH_SIZE_WARNING.call_once(|| {
+                        let size_mb = size_bytes / 1024 / 1024;
+                        warn!("Lance read in a single batch that contained more than {}MiB of data.  You may want to consider reducing the batch size.", size_mb);
+                    });
```

Comment:
> Do you worry this could be noisy? Would someone with image / video datasets always hit this?

10MiB is pretty extreme but yes, someone with a video dataset would encounter this even if there was only one video.  Perhaps I will only emit the warning if the batch size is greater than 32 (just some arbitrary threshold).

> Also, does this ever reset? To try again with a smaller batch size and see if they get this warning, will they need to restart the process?

Hmm, I was thinking "once per scan" but, on reflection, it seems I ended up with "once per process".  I will fix.

Reply:
Changed to once-per-scan.